### PR TITLE
[Fix clojure-emacs/cider#1474] Add a debug key for printing the stack

### PR DESCRIPTION
**Pull request by Malabarba**

### CODE REVIEW

**Comment by bbatsov** [comment]
Apart from the failing tests - looks good to me.

File: src/cider/nrepl/middleware/debug.clj [hunk 1]
```diff
@@ -210,6 +210,29 @@
                                                    (inspect/start inspect-value)))))]
     (read-debug-command value (assoc extras :inspect i))))
 
+(defn flag-debug-middleware
```
Comment (bbatsov) [inline src/cider/nrepl/middleware/debug.clj]: Btw, we can probably do this for all `cider-nrepl` namespaces.

File: src/cider/nrepl/middleware/debug.clj [hunk 2]
```diff
@@ -221,6 +244,7 @@
     locals: Inspect local variables.
     inject: Evaluate an expression and return it.
     eval: Evaluate an expression, display result, and prompt again.
+    stack: Print the current stack, and prompt again.
```
Comment (bbatsov) [inline src/cider/nrepl/middleware/debug.clj]: Maybe we should name this `stacktrace` to be crystal clear what exactly the command is for.

**Comment by bbatsov** [comment]
You'll have to rebase this branch on top of the current `master`.

**Comment by Malabarba** [comment]
Yeah, I still have to improve the flagging function too.

**Comment by Malabarba** [comment]
Done. But please have a second look, as I've moved the flagging change straight to `stacktrace.clj`.

**Comment by bbatsov** [comment]
The changes look good to me.

**Pull request merged**